cnitool: deduplicate environment variable definitions

UiP9AV6Y opened the pull request:
create a reusable package for environment variable to be used by `cnitool` and the other library code.

Signed-off-by: Gordon Bleux <33967640+UiP9AV6Y@users.noreply.github.com>

squeed commented on pkg/env/env.go:
```diff
@@ -0,0 +1,95 @@
+
+	DefaultCapabilityArgs = ""
+	DefaultCNIArgs        = ""
+	DefaultCNIIfname      = "eth0"
```
This should probably be the empty string

squeed commented on pkg/env/env.go:
```diff
@@ -0,0 +1,95 @@
+	DefaultCapabilityArgs = ""
+	DefaultCNIArgs        = ""
+	DefaultCNIIfname      = "eth0"
+	DefaultNetDir         = "/etc/cni/net.d"
```
This should be the empty string too.

squeed commented on pkg/env/env.go:
```diff
@@ -0,0 +1,95 @@
+
+// GetNetDir gets the network configuration
+// directory setting from the environment
+func GetNetDir() string {
```
Should this be in a central package? It's only used by cnitool, right?

squeed commented:
This looks basically reasonable, but any of the environment variables that are `cnitool`-specific should probably not be in `pkg`. That's just because we don't want to make any public library commitments in that regard.

squeed commented on pkg/skel/skel.go:
```diff
@@ -65,77 +66,54 @@ func (t *dispatcher) getCmdArgsFromEnv() (string, *CmdArgs, *types.Error) {
-				"CHECK": true,
-				"DEL":   true,
-			},
+			newReqForCmdEntry(true, true, true),
```
nit: I prefer the literal here, if you don't mind.

UiP9AV6Y commented:
i have moved all `cnitool` specific default values back into
the cmd package to prevent having too many *empty defaults*
(the default has to be defined by the consumer when using 
`GetVal()`)

furthermore i have moved the CNI Args parsing logic into
the library API as this is something covered by the
[SPEC](https://github.com/containernetworking/cni/blob/e629d106b4e02030d0f7794758b0f0438b6a78dd/SPEC.md?plain=1#L209)
